### ISSUE TITLE
SwiftUI PaymentButton crash

### ISSUE THREAD

**Issue by tnasyrov-sphereinc** [stripe/stripe-ios#1900]
## Summary
I'm working on a SwiftUI app, and everything was working fine for a long time, haven't changed any code regarding the Stripe payments, but after some of the latest SDK update I suddenly get a crash when a user tries to open the payment sheet with a `PaymentSheet.PaymentButton`. Here's an error text I get in the console: 
`Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Application tried to present modally a view controller <Stripe.BottomSheetViewController: 0x7fb3bbdb1890> that is already being presented by <_TtGC7SwiftUI29PresentationHostingControllerVS_7AnyView_: 0x7fb3bbca0700>.'`
I would really appreciate any tips on why this is happening and how it could be fixed!

## Code to reproduce
The code is really simple, I just use the PaymentButton to present the payment sheet:
```
if let paymentSheet = viewModel.paymentSheet {
  PaymentSheet.PaymentButton(
      paymentSheet: paymentSheet,
      onCompletion: { result in
          viewModel.onStripePaymentCompletion(result: result)
      }
  ) {
      Spacer()
      Text(Texts.tapToBook(viewModel.totalPrice))
      Spacer()
  }
}
```

## iOS version
Tested on iOS 14.4 (real device) and iOS 15.0 (simulator)

## Installation method
SPM

## SDK version
21.9.0

## Other information
Call stack:
```
	0   CoreFoundation                      0x00000001087ffbb4 __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x0000000104e10be7 objc_exception_throw + 48
	2   UIKitCore                           0x0000000120bfca52 -[UIViewController _presentViewController:withAnimationController:completion:] + 5854
	3   UIKitCore                           0x0000000120bfd520 __63-[UIViewController _presentViewController:animated:completion:]_block_invoke + 98
	4   UIKitCore                           0x0000000120c1898c -[_UIViewControllerTransitionCoordinator _applyBlocks:releaseBlocks:] + 294
	5   UIKitCore                           0x0000000120c14cbe -[_UIViewControllerTransitionContext _runAlongsideCompletions] + 126
	6   UIKitCore                           0x0000000120c14a52 -[_UIViewControllerTransitionContext completeTransition:] + 117
	7   Roomer                              0x00000001020b2aa9 $s6Stripe28PanModalPresentationAnimatorC07animateD033_726A1B5B76B88C60626FE8B5355A436ALL17transitionContextySo016UIViewControllerP13Transitioning_p_tFySbcfU0_ + 169
	8   Roomer                              0x00000001020b1817 $s6Stripe16PanModalAnimatorV7animate_6config_yyyc_AA0bC11Presentable_pSgySbcSgtFZySo23UIViewAnimatingPositionVcfU_ + 55
	9   Roomer                              0x00000001020b18b0 $sSo23UIViewAnimatingPositionVIegy_ABIeyBy_TR + 48
	10  UIKitCore                           0x000000012078b8a5 -[UIViewPropertyAnimator _executeCompletionHandlerWithFinalPosition:] + 226
	11  UIKitCore                           0x000000012078b99a -[UIViewPropertyAnimator _runCompletions:finished:] + 118
	12  UIKitCore                           0x000000012078a68b __61-[UIViewPropertyAnimator _setupAssociatedViewAnimationState:]_block_invoke + 168
	13  UIKitCore                           0x000000012195758a -[UIViewAnimationBlockDelegate _didEndBlockAnimation:finished:context:] + 779
	14  UIKitCore                           0x0000000121927f81 -[UIViewAnimationState sendDelegateAnimationDidStop:finished:] + 231
	15  UIKitCore                           0x0000000121928526 -[UIViewAnimationState animationDidStop:finished:] + 263
	16  QuartzCore                          0x00000001084429b2 _ZN2CA5Layer23run_animation_callbacksEPv + 308
	17  libdispatch.dylib                   0x0000000109d97c0c _dispatch_client_callout + 8
	18  libdispatch.dylib                   0x0000000109da6376 _dispatch_main_queue_callback_4CF + 1195
	19  CoreFoundation                      0x000000010876d85d __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	20  CoreFoundation                      0x00000001087680ba __CFRunLoopRun + 2772
	21  CoreFoundation                      0x0000000108767103 CFRunLoopRunSpecific + 567
	22  GraphicsServices                    0x000000010d5dbcd3 GSEventRunModal + 139
	23  UIKitCore                           0x00000001213a2e63 -[UIApplication _run] + 928
	24  UIKitCore                           0x00000001213a7a53 UIApplicationMain + 101
	25  libswiftUIKit.dylib                 0x0000000109c7a052 $s5UIKit17UIApplicationMainys5Int32VAD_SpySpys4Int8VGGSgSSSgAJtF + 98
	26  Roomer                              0x00000001017e3b68 $sSo21UIApplicationDelegateP5UIKitE4mainyyFZ + 104
	27  Roomer                              0x00000001017e3af7 $s6Roomer11AppDelegateC5$mainyyFZ + 39
	28  Roomer                              0x00000001017e3c78 main + 24
	29  dyld                                0x0000000104ab1e1e start_sim + 10
	30  ???                                 0x00000001091294fe 0x0 + 4447180030
	31  ???                                 0x0000000000000000 0x0 + 0
	32  Roomer                              0x00000001017b0000 __dso_handle + 0
```

**Comment by tnasyrov-sphereinc** [comment]
I've just tried to roll back to Stripe 21.8.1 and everything's working just fine again.

**Comment by csabol-stripe** [comment]
Hmm, I just tried reproducing this in our [PaymentSheet Example app](https://github.com/stripe/stripe-ios/blob/39abf35e84f2e284aabfa9ad9efc7ce6a13c5f1c/Example/PaymentSheet%20Example/PaymentSheet%20Example/PaymentSheet%20Example/ExampleSwiftUIPaymentSheet.swift) and didn't get a crash. Could you share a code sample that you're able to reproduce with? Or you can reach out to Stripe support and we can take a look at your integration :) https://support.stripe.com/contact/login

**Comment by codeonomics** [comment]
Also getting this issue but presenting via the modifier inside a navigation stack. https://github.com/stripe/stripe-ios/issues/1906

**Comment by csabol-stripe** [comment]
@codeonomics I'm taking another look at this with some additional reproduction steps we've been able to come up with

**Comment by csabol-stripe** [comment]
Got a fix for this merged and we'll include it in our next release!

**Comment by rizkicalame** [comment]
Hey @csabol-stripe!

Any ideas on when a release is planned for this fix? A big portion of our userbase is affected by this. Perhaps a hotfix with this fix included might be a good idea?

**Comment by porter-stripe** [comment]
Closing as this should be resolved in our latest 21.12.0 release
